### PR TITLE
construct client with optional credentials

### PR DESCRIPTION
**Pull request by delocalizer**
This is certainly useful for us, perhaps for others.

### CODE REVIEW

**Comment by delocalizer** [comment]
As background, we run cromwell server behind an authenticating proxy.

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala [hunk 1]
```diff
@@ -19,7 +21,10 @@ import cromwell.api.CromwellClient._
 
 import scala.util.{Failure, Success, Try}
 
-class CromwellClient(val cromwellUrl: URL, val apiVersion: String)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) {
+class CromwellClient(val cromwellUrl: URL, val apiVersion: String, val credentials: Option[HttpCredentials]=None)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) {
+
+  lazy val authHeader = credentials.map { Authorization(_) }
+  lazy val commonHeaders = authHeader.toList
```
Comment (geoffjentry) [inline cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: Why not just `toList` it up above? This is only being used in the one place down below, not sure I understand the multiple layers here.

**Comment by geoffjentry** [comment]
👍 - I suspect we'll be running into this soon as well, thanks.

**Comment by cjllanwarne** [comment]
Very cool! I know that we'd have needed this soon too so really helpful!

**Comment by cjllanwarne** [comment]
👍

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala (hunk 1 above)
Comment (cjllanwarne) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: I could be persuaded that splitting it up like this makes things more readable/understandable and the compiler will probably inline it anyway if it gets called a lot. So, I'd be happy either way.

Comment (geoffjentry) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: Don't completely disagree w/ that but my main question was having two basically the same variables at the top level which ultimately are only used in the one function. And really the question part of my main question was if there was a reason for that setup :)

Comment (delocalizer) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: Yeah, I thought for clarity and a little future-proofing that `executeRequest` should add all common headers, and creating the List of them should be separate from creating each one of them from possibly optional pieces.

Comment (geoffjentry) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: I was almost going to suggest inlining the whole thing in the one spot so you can see where my biases lie :) Not a big deal, carry on. Will merge this, thanks @delocalizer

**Pull request merged**